### PR TITLE
Update control owning field  using weak reference.(#9452) simplify code

### PR DESCRIPTION
**Pull request by Epica3055**
Related to #9224 

## Proposed changes

- Simplify code

## Regression? 

-  No

## Test methodology 

- Unit tests
 

## Test environment(s) 

- dotnet 8.0.0-preview.7.23361.9

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/9485)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/OwnedObjectExtensions.cs [hunk 1]
```diff
@@ -18,4 +18,17 @@ internal static class OwnedObjectExtensions
         ownerAs = ownedObject.Owner as TAs;
         return ownerAs is not null;
     }
+
+    /// <summary>
+    ///  Tries to get IsHandleCreated from the owner.
+    /// </summary>
+    public static bool IsHandleCreated<TOwner, TAs>(
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/OwnedObjectExtensions.cs:25]: Consider `IsOwnerHandleCreated`

File: src/System.Windows.Forms/src/System/Windows/Forms/OwnedObjectExtensions.cs [hunk 2]
```diff
@@ -18,4 +18,17 @@ internal static class OwnedObjectExtensions
         ownerAs = ownedObject.Owner as TAs;
         return ownerAs is not null;
     }
+
+    /// <summary>
+    ///  Tries to get IsHandleCreated from the owner.
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/OwnedObjectExtensions.cs:23]:
```suggestion
    ///  If the owner object exists, then verify if the handle is created, if owner does not exist, return <see  langword="false" />.
```

File: src/System.Windows.Forms/src/System/Windows/Forms/OwnedObjectExtensions.cs [hunk 3]
```diff
@@ -18,4 +18,17 @@ internal static class OwnedObjectExtensions
...
+    /// </summary>
+    public static bool IsHandleCreated<TOwner, TAs>(
+        this IOwnedObject<TOwner> ownedObject,
+        [NotNullWhen(true)] out TAs? ownerAs)
+        where TOwner : class
+        where TAs : Control
+    {
+        ownerAs = ownedObject.Owner as TAs;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/OwnedObjectExtensions.cs:31]: Did you want to set ownerAs to null when Handle is not created to match what the NotNullWhen attribute?

**Review by Tanya-Solyanik — COMMENTED** [review]
Added some comments, but I didn't finish the review

**Comment by elachlan** [comment]
There are quite a few places where you output `owner` but instead should use a discard local `_` because owner isn't used afterwards.

**Review by dmitrii-drobotov — APPROVED** [review]

**Review by Tanya-Solyanik — APPROVED** [review]
Thank you!

**Pull request merged**